libs/datasets/timeseries.py

```diff
@@ -634,21 +649,33 @@ def timeseries_bucketed_long(self) -> pd.Series:
         """A Series with MultiIndex LOCATION_ID, DEMOGRAPHIC_BUCKET, DATE, VARIABLE"""
         return self.timeseries_bucketed.stack(dropna=True).rename(PdFields.VALUE).sort_index()
 
+    @cached_property
+    def timeseries_distribution_long(self) -> pd.Series:
+        """A Series with MultiIndex LOCATION_ID, DEMOGRAPHIC_BUCKET, DATE, DISTRIBUTION, VARIABLE"""
+        return (
+            _add_distribution_level(self.timeseries_bucketed)
+            .stack(dropna=True)
+            .rename(PdFields.VALUE)
+            .sort_index()
+        )
+
     @cached_property
     def wide_var_not_null(self) -> pd.DataFrame:
         """True iff there is at least one real value in any bucket with given location and
         variable"""
         wide_var_not_null = (
-            self.timeseries_bucketed_long.notnull()
-            .groupby([CommonFields.LOCATION_ID, PdFields.VARIABLE], sort=False)
+            self.timeseries_distribution_long.notna()
```

Comment:
Are we intentionally switching from `notnull()` to `notna()`. Is it safe to assume that we never have `None` or `NaT` and so this is equivalent?

Reply:
`notnull` and `notna` are aliases. https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.notna.html and thanks to https://github.com/covid-projections/covid-data-model/blob/main/libs/datasets/timeseries.py#L993 we can be sure all values are floats (real or nan).